test-poc/examples/src/test/java/org/keycloak/test/examples/WelcomePageTest.java

```diff
@@ -21,18 +34,91 @@ public class WelcomePageTest {
     @InjectAdminClient
     Keycloak adminClient;
 
+    @InjectPage
+    WelcomePage welcomePage;
+
     @Test
-    public void testCreateUser() {
-        final var welcomePage = new WelcomePage(driver);
+    @Order(1)
+    public void localAccessNoAdmin() throws Exception {
+        welcomePage.navigateTo();
+        Assertions.assertFalse(welcomePage.isPasswordSet(), "Welcome page did not ask to create a new admin user.");
+    }
+
+    @Test
+    @Order(2)
+    public void remoteAccessNoAdmin() throws Exception {
+        driver.get(getPublicServerUrl().toString());
+        Assertions.assertFalse(welcomePage.isPasswordSet(), "Welcome page did not ask to create a new admin user.");
+    }
+
+    @Test
+    @Order(3)
+    public void createAdminUser() {
         welcomePage.navigateTo();
         welcomePage.fillRegistration("admin", "admin");
         welcomePage.submit();
         welcomePage.assertUserCreated();
 
         List<UserRepresentation> users = adminClient.realm("master").users().search("admin", true);
         Assertions.assertEquals(1, users.size());
+    }
+
+    @Test
+    @Order(4)
+    public void localAccessWithAdmin() throws Exception {
+        welcomePage.navigateTo();
+        Assertions.assertTrue(welcomePage.isPasswordSet(), "Welcome page asked to set admin password.");
+    }
 
-        adminClient.realm("master").users().get(users.get(0).getId()).remove();
+    @Test
+    @Order(5)
+    public void remoteAccessWithAdmin() throws Exception {
+        driver.get(getPublicServerUrl().toString());
+        Assertions.assertTrue(welcomePage.isPasswordSet(), "Welcome page asked to set admin password.");
     }
 
+    @Test
+    @Order(6)
+    public void accessCreatedAdminAccount() throws Exception {
+        welcomePage.navigateTo();
+        welcomePage.navigateToAdminConsole();
+        Assertions.assertEquals(driver.getTitle(), "Keycloak Administration Console");
+    }
+
+    @Test
+    @Order(7)
+    public void checkProductNameOnWelcomePage() {
+        welcomePage.navigateTo();
+        welcomePage.login("admin", "admin");
+        String actualMessage = welcomePage.getWelcomeMessage();
+        Assertions.assertEquals("Welcome to Keycloak", actualMessage);
+    }
+
+    /**
+     * Attempt to resolve the floating IP address. This is where EAP/WildFly
+     * will be accessible. See "-Djboss.bind.address=0.0.0.0".
```

Comment:
@wojnarfilip I think these lines are outdated -> right now we use Quarkus

Reply:
Updated